OpenTaiko/src/Common/ImGuiDebugWindow.cs

```diff
@@ -346,12 +349,7 @@ private static void Stage() {
 		}
 		private static void Textures() {
 			if (ImGui.BeginTabItem("Textures")) {
```

Comment:
Could all be replaced by early-return or even better create a switch that checks if the method should be called since it's not called somewhere else.

Reply:
The block nesting reflects the UI layout hierarchy. Applying early-return here might make the code less intuitive.

Reply:
I'm not sure if I understand correctly why the early-return would make it less intuitive. **Yes** it does reflect UI layout hierarchy, but why should the following be less intuitive?
```csharp
private static void Textures() {
  if (!ImGui.BeginTabItem("Textures")) {
    return;
  }
  // Rest of the code
```
I would only find the current version more useful for UI layout if `ImGui.EndTabItem();` wouldn't always be called at the end. If we rename `Textures` with `TextureTab` the code should still be readable with early return pattern and we would profit from less indentation. But I'm open to other suggestions!

Reply:
The factor seems to be resource management.
`ImGui.BeginTabItem()` has a property of "cannot revert automatically" and must be reverted using `ImGui.EndTabItem()`.
Brace nesting can serve as a visual hint of "still holding the resource" (here unended tab item) and a reminder to call the reverting method.
A better pattern is to have an `IDisposable` API, and use something like `using (ImGui.TabItem()) { }` ※, or in this case (resource is sole in its level) `using var _ = ImGui.TabItem();` ※ at the head of the method. This way the reverting method is called even if the level is exited by an uncaught exception.
※ fictional method

Reply:
Since the visual hint for me is the `method` (and not the `if`) I'm not sure if I would agree on the first part in this specific case. As I noted it would make more sense to do this in cases where `BeginTabItem` and `EndTabItem` isn't always called at the beginning and ending. But even then it would probably make sense to split this into smaller parts.

Otherwise I agree with you that we should be using the `IDisposable` API. If the nesting is happening because of a `using ... { }` that would be fine for me. I'm a bit rusty in `C#`, but I think we could even reduce nesting with your second suggestion `using var`. But that's not needed, I think tho that we should use the `IDisposable` API for the purpose of proper cleanup. We might open up a subsequent PR for this.

Reply:
`true`, I now see the pattern of the user and reverter on respectively the first/leading and last line of a block. I must have been afraid of hypothetical mid-block users.